src/realm/sync/client.cpp

```diff
@@ -1701,28 +1708,28 @@ void SessionWrapper::report_progress(ReportedProgress& p, DownloadableProgress d
         upload_estimate = calculate_progress(p.uploaded, p.uploadable, m_final_uploaded);
 
     bool download_completed = p.downloaded == 0;
-    double download_estimate = 1.00;
+    p.download_estimate = 1.00;
```

Comment:
that check below should be also fixed to take into account initial history scan case where changes are applied directly without hitting bootstrap store first

i think it should just check that this is the flx connection and set estimate unconditionally since it's always provided

Reply:
Good catch. I wrote up a quick test case that demonstrates the issue:
```
    SECTION("history scan") {
        config.sync_config->subscription_initializer = [](const std::shared_ptr<Realm>& realm) {
            subscribe_to_all(*realm);
        };

        // Open a realm and bootstrap it so that we are getting progress notifications for history
        // scans rather than bootstraps.
        {
            successfully_async_open_realm(config);
        }

        // In another realm, add some more objects to the server that will need to be downloaded
        // in a bunch of changesets.
        harness->do_with_new_realm([](const std::shared_ptr<Realm>& realm) {
            const size_t padding_size = 1024 * 1024;
            auto buffer = std::make_unique<char[]>(padding_size);
            auto table = realm->read_group().get_table("class_object");
            subscribe_to_all(*realm);
            for (int i = 5; i < 10; ++i) {
                realm->begin_transaction();
                auto obj = table->create_object_with_primary_key(ObjectId::gen());
                obj.set("int", i);
                // ensure that each object is large enough that it'll be sent in
                // a separate DOWNLOAD message
                obj.set("padding", BinaryData(buffer.get(), padding_size));
                realm->commit_transaction();
            }
            wait_for_upload(*realm);
        });

        // Re-open the original realm and wait for it to download everything. We should see
        // a new batch of progress esimates that are also monotonically increasing.
        auto task = Realm::get_synchronized_realm(config);
        std::vector<double> estimates;
        task->register_download_progress_notifier([&](uint64_t, uint64_t, double estimate) {
            // Note that no locking is needed here despite this being called on
            // a background thread as the test provides the required synchronization.
            // If tsan complains about this, it indicates that the notifier is
            // being called at a time that it shouldn't be and there's a bug.
            estimates.push_back(estimate);
        });
        task->start().get();
        REQUIRE(estimates.size() >= 5);
        REQUIRE_THAT(estimates, EstimatesAreValid());
    }
```